rdt/hyper_transformer.py

```diff
@@ -335,6 +375,8 @@ def update_transformers_by_sdtype(
         self._warn_update_transformers_by_sdtype(transformer, transformer_name)
 
         transformer_instance = transformer
+        self._multi_column_fields = self._create_multi_column_fields()
```

Comment:
we should just remove this from the init and move it into the detect method. Then we don't need to keep calling it